notebooks/working/4.0-mab-causal-graph-simulation.md

```diff
@@ -0,0 +1,286 @@
+BATS_IND = BATS_IND[(BATS_IND.SCHTYP!=998)&(BATS_IND.SCHTYP!=999)]
+BATS_IND = BATS_IND[(BATS_IND.DISTYPE!=998)&(BATS_IND.DISTYPE!=999)]
+BATS_IND = BATS_IND[(BATS_IND.ETHNIC!=998)&(BATS_IND.ETHNIC!=999)].reset_index(drop=True)
+
+BATS_IND['income'] = BATS_IND.HHINCOME.apply(
+                                            lambda x: random.randrange(0,15000,10)if x==1 else
+                                                     (random.randrange(10000,15000,10)if x==2 else 
+                                                     (random.randrange(15000,20000,10)if x==3 else
+                                                     (random.randrange(20000,25000,10)if x==4 else
+                                                     (random.randrange(25000,30000,10)if x==5 else
+                                                     (random.randrange(30000,35000,10)if x==6 else
+                                                     (random.randrange(35000,40000,10)if x==7 else
+                                                     (random.randrange(40000,45000,10)if x==8 else
+                                                     (random.randrange(45000,50000,10)if x==9 else
+                                                     (random.randrange(50000,60000,10)if x==10 else
+                                                     (random.randrange(60000,75000,10)if x==11 else
+                                                     (random.randrange(75000,100000,10)if x==12 else
+                                                     (random.randrange(100000,125000,10)if x==13 else
+                                                     (random.randrange(125000,150000,10)if x==14 else
+                                                      random.randrange(150000,250000,10)))))))))))))))
+BATS_IND = BATS_IND[BATS_IND['income']!= 0]
+```
+
+## Generate Variables 
+
+```python
+age = BATS_IND.AGE.values
+income = BATS_IND.income.values
+income_div = income/100000
+tt_DA = np.abs(np.sqrt(age) + 500000/income + np.random.normal(0, 1, BATS_IND.shape[0]))
+tt_TR = np.abs(2.5*age + 40000/np.sqrt(income) + np.random.normal(0, 1, BATS_IND.shape[0]))
+tt_BS = np.abs(1.5*age + 500000/income + np.random.normal(0, 1, BATS_IND.shape[0]))
+cost_DA = np.abs(10*tt_DA + np.random.normal(0, 1, BATS_IND.shape[0]))
+cost_TR = np.abs(tt_TR/100 + np.random.normal(0, 1, BATS_IND.shape[0]))
+cost_BS = np.abs(0.25*tt_BS + np.random.normal(0, 1, BATS_IND.shape[0]))
+```
+
+## Generate dataset based on assumed Causal Graph
+
+```python
+sample_data = CreateChoiceDF(("age", age),
+                             ("income", income_div),
+                             ("tt_DA", tt_DA),
+                             ("tt_BS", tt_BS),
+                             ("tt_TR", tt_TR),
+                             ("cost_DA", cost_DA),
+                             ("cost_BS", cost_BS),
+                             ("cost_TR", cost_TR)
+                            )
+```
+
```

Comment:
@bouzaghrane @timothyb0912 So if I understand correctly, the relationship between the explanatory variables in the causal are specified here right? The form of these relationships is determined by running distributional regressions separately.

Reply:
Yes sir. I am working on the workflow for getting these simulations done based on each alternative separately, since we technically will have 8 alternatives and therefore 8 different simulations. But you are right!